modules/build/src/main/scala/scala/build/Sources.scala

```diff
@@ -86,14 +86,22 @@ object Sources {
     generatedRelPath: os.RelPath,
     wrapScriptFun: CodeWrapper => (String, WrapperParams)
   ) {
-    def wrap(wrapper: CodeWrapper): InMemory = {
+    def wrap(wrapper: CodeWrapper): Seq[InMemory] = {
       val (content, wrapperParams) = wrapScriptFun(wrapper)
-      InMemory(
-        originalPath,
-        generatedRelPath,
-        content.getBytes(StandardCharsets.UTF_8),
-        Some(wrapperParams)
-      )
+      Seq(
+        InMemory(
+          originalPath,
+          generatedRelPath,
+          content.getBytes(StandardCharsets.UTF_8),
+          Some(wrapperParams))
+      ) ++ wrapper.additionalSourceCode.map { sourceCode =>
+        InMemory(
+          Left("script-wrapper"),
+          generatedRelPath / os.up / "script-wrapper.scala",
```

Comment:
might want to make this path unique per-script.
maybe smth like:
```suggestion
          generatedRelPath / os.up / s"${generatedRelPath.last.replace(".", "_")}_script-wrapper.scala",
```

Reply:
RIght now all the script-wrapper.scala files contain the same code, so naming them the same allows for filtering them out with `.distinct`. Although it would be good to name them per-wrapper so we could have different code generated for class and object wrappers.